waku/v2/peerstore/waku_peer_store.go

```diff
@@ -139,3 +145,43 @@ func (ps *WakuPeerstoreImpl) Direction(p peer.ID) (network.Direction, error) {
 
 	return result.(network.Direction), nil
 }
+
+// AddPubSubTopic adds a new pubSubTopic for a peer
+func (ps *WakuPeerstoreImpl) AddPubSubTopic(p peer.ID, topic string) error {
+	existingTopics, err := ps.PubSubTopics(p)
+	if err != nil {
+		return err
+	}
+	existingTopics = append(existingTopics, topic)
+	return ps.peerStore.Put(p, peerPubSubTopics, existingTopics)
+}
+
+// SetPubSubTopics sets pubSubTopics for a peer, it also overrides existing ones that were set previously..
+func (ps *WakuPeerstoreImpl) SetPubSubTopics(p peer.ID, topics []string) error {
+	return ps.peerStore.Put(p, peerPubSubTopics, topics)
+}
+
+// PubSubTopics fetches list of pubSubTopics for a peer
+func (ps *WakuPeerstoreImpl) PubSubTopics(p peer.ID) ([]string, error) {
```

Comment:
Would it make sense to check the shards stored in the ENRs of peers retrieved via discv5/peer exchange, and return these as well? (this question applies to `PeersByPubSubTopic` too)

Reply:
Those are already being translated to pubSubTopics and added when the peer is added to peerStore. 
https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/peermanager/peer_manager.go#L219

https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/discv5/discover.go#L492-L498

and 
https://github.com/waku-org/go-waku/blob/6f8933ef7971e83987f53f07b8ff89423ec964b2/waku/v2/protocol/peer_exchange/client.go#L111-L120

Unless i am missing something.

Reply:
oh i see. I missed this!